### PR TITLE
apply the custom probe configs when externalTrafficPolicy is local and PLSProxyProtocol is enabled.

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
apply the custom probe configs when externalTrafficPolicy is local and PLSProxyProtocol is enabled.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
apply the custom probe configs when externalTrafficPolicy is local and PLSProxyProtocol is enabled.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_loadbalancer.go [hunk 1]
```diff
@@ -2381,7 +2381,7 @@ func (az *Cloud) getExpectedLBRules(
 	// healthcheck proxy server serves http requests
 	// https://github.com/kubernetes/kubernetes/blob/7c013c3f64db33cf19f38bb2fc8d9182e42b0b7b/pkg/proxy/healthcheck/service_health.go#L236
 	var nodeEndpointHealthprobe *network.Probe
-	if servicehelpers.NeedsHealthCheck(service) {
+	if servicehelpers.NeedsHealthCheck(service) && !consts.IsPLSProxyProtocolEnabled(service.Annotations) {
```
Comment (jwtty) [inline pkg/provider/azure_loadbalancer.go:2384]: Can we also check if service.beta.kubernetes.io/azure-pls-create is true? want to make sure pls is indeed needed here?

Comment (MartinForReal) [reply pkg/provider/azure_loadbalancer.go]: updated. Thanks!

File: pkg/provider/azure_loadbalancer.go [hunk 2]
```diff
@@ -2381,7 +2381,7 @@ func (az *Cloud) getExpectedLBRules(
 	// healthcheck proxy server serves http requests
 	// https://github.com/kubernetes/kubernetes/blob/7c013c3f64db33cf19f38bb2fc8d9182e42b0b7b/pkg/proxy/healthcheck/service_health.go#L236
 	var nodeEndpointHealthprobe *network.Probe
-	if servicehelpers.NeedsHealthCheck(service) {
+	if servicehelpers.NeedsHealthCheck(service) && !(consts.IsPLSEnabled() && consts.IsPLSProxyProtocolEnabled(service.Annotations)) {
```
Comment (jwtty) [inline pkg/provider/azure_loadbalancer.go:2384]: nit: arg missing

**Pull request merged**